src/plugins/controls/public/react_controls/control_group/init_controls_manager.ts

```diff
@@ -222,3 +234,16 @@ export function initControlsManager(initialControlPanelsState: ControlPanelsStat
     >,
   };
 }
+
+function getLastUsedDataViewId(initialControlPanelsState: ControlPanelsState) {
```

Comment:
nit: do we actually want to get the _first_ used data view id? Or do we want the most recently used data view id?

A closer proxy for _most recently used_ might be the data view used by the control closest to the end of the chain (assuming folks add controls one by one). In that case, looping backwards might be better.

Either way, just a tiny nit.